### PR TITLE
Provide a way to enter a line total and have the quantity sold automatically calculated

### PR DESCRIPTION
**Pull request by SteveIreland**
This pull request addresses the issue reported in #1647

### CODE REVIEW

**Comment by daN4cat** [comment]
I fixed few things directly to avoid the review turnaround.

Thanks @SteveIreland for the work and contribution and Happy New Year!

**Comment by odiea** [comment]
@SteveIreland  What happened to this line of code. It appears to be missing in the Register.
```							
<td><?php echo to_currency($item['price']*$item['quantity']-$item['price']*$item['quantity']*$item['discount']/100); ?></td>

**Comment by daN4cat** [comment]
Indeed just noticed that there is an issue in the register now. Just applying a discount the total doesn't change.

**Comment by daN4cat** [comment]
Fixing

**Comment by odiea** [comment]
I think this if should be updated to include if this app config is enabled. Currently the item still allows the change when just this is encluded.
```
if($item['item_type'] == 2)
									{
									echo form_input(array('name'=>'total', 'class'=>'form-control input-sm', 'value'=>to_currency_no_money($item['total']), 'tabindex'=>++$tabindex));
									}

**Comment by daN4cat** [comment]
Fixed and in master

**Comment by odiea** [comment]
@SteveIreland If you get a spare minute can you explain the following. Occasionally I get an undefined error.  ITEM_PRICE_BASED. Models/item.php
```
	public function get_stock_search_suggestions($search, $filters = array('is_deleted' => FALSE, 'search_custom' => FALSE), $unique = FALSE, $limit = 25)
	{
		$suggestions = array();
		$non_kit = array(ITEM, ITEM_PRICE_BASED);

**Comment by daN4cat** [comment]
Is ITEM_PRICE_BASED defined? How can you rrproduce this error?

**Comment by odiea** [comment]
I only get this when I use Receivings. It's possible I had a bad merg but I can't see where this is defined anywhere.
ERROR - 2018-01-31 06:12:56 --> Severity: Notice --> Use of undefined constant ITEM_PRICE_BASED - assumed 'ITEM_PRICE_BASED' C:\xampp\htdocs\featuremaster3\application\models\Item.php 628

**Comment by daN4cat** [comment]
I see it's used in `get_stock_search_suggestions` and `get_kit_search_suggestions`.

@SteveIreland what was the aim there? I can't find it defined....

File: application/models/Item.php [hunk 1]
```diff
@@ -598,11 +600,12 @@ public function get_search_suggestions($search, $filters = array('is_deleted' =>
 	public function get_stock_search_suggestions($search, $filters = array('is_deleted' => FALSE, 'search_custom' => FALSE), $unique = FALSE, $limit = 25)
 	{
 		$suggestions = array();
+		$non_kit = array(ITEM, ITEM_PRICE_BASED);
```
Comment (daN4cat) [inline application/models/Item.php]: @SteveIreland ITEM_PRICE_BASED is not defined...

**Comment by daN4cat** [comment]
I'm not sure @SteveIreland is still watching this space....

**Pull request merged**